src/strategies/filter-music-nfts/extractor.mjs

```diff
@@ -0,0 +1,77 @@
+// @format
+import { createInterface } from "readline";
+import { createReadStream } from "fs";
+import { decodeLog } from "eth-fun";
+
+import logger from "../../logger.mjs";
+import { parseJSON } from "../../utils.mjs";
+
+export const name = "filter-music-nfts";
+const log = logger(name);
+export const version = "0.1.0";
+export const props = {};
+
+function decodeArtistAddress(log) {
+  const topics = log.topics;
+  topics.shift();
+  const result = decodeLog(
+    [
+      {
+        type: "uint256",
+        name: "artistId",
+      },
+      {
+        type: "string",
+        name: "name",
+      },
+      {
+        type: "string",
+        name: "symbol",
+      },
+      {
+        type: "address",
+        name: "artistAddress",
+        indexed: true,
+      },
+    ],
+    log.data,
+    topics
+  );
+  return result.artistAddress.toLowerCase();
+}
+
+export async function init(logPath, contracts) {
```

Comment:
we must not abuse the extractor interface for filtering

Reply:
Should we use transformer interface for filtering? My thought was that it is extracting a subset of data from the larger dataset.

Reply:
@il3ven I'd like to understand: What stopped you from using the transformer interface to handle this type of task? If we need a "filter" strategy interface (that is different from transform and extraction, we should consider adding it.

Reply:
> Should we use transformer interface for filtering? My thought was that it is extracting a subset of data from the larger dataset.

mhh language-wise that makes sense to me but I I think the extractor is supposed to be used to download data from the internet.

Reply:
> What stopped you from using the transformer interface to handle this type of task?

Nothing. language-wise it seemed natural to use extractor. I am fine with transformer too.